fea-rs/src/bin/compile.rs

```diff
@@ -60,47 +47,37 @@ fn main() {
             println!("{} errors, {} warnings", err_count, warning_count);
             std::process::exit(1);
         }
-    }
+    };
 
-    //match &args.subcommand {
-    //flags::ArgsCmd::Compile(args) => {
-    //if let Some(path) = &args.out_path {
-    //font.save(path).unwrap()
-    //} else {
-    //font.save("compile-out.ttf").unwrap()
-    //}
-    //}
-    //flags::ArgsCmd::Debug(args) => {
-    //let to_print = args
-    //.print_tables
-    //.as_ref()
-    //.map(|s| s.split(',').map(|s| s.to_owned()).collect::<HashSet<_>>())
-    //.unwrap_or_default();
-    //if to_print.is_empty() {
-    //fea_rs::util::debug::explode_font(&font, args.verbose);
-    //}
-
-    //for table in to_print {
-    //if table == "GPOS" {
-    //if let Some(gpos) = font.tables.GPOS().unwrap() {
-    //util::debug::explode_gpos(&gpos, args.verbose);
-    //} else {
-    //eprintln!("no GPOS table exists");
-    //}
-    //} else if table == "GSUB" {
-    //if let Some(gsub) = font.tables.GSUB().unwrap() {
-    //util::debug::explode_gsub(&gsub, args.verbose);
-    //} else {
-    //eprintln!("no GSUB table exists");
-    //}
-    //} else {
-    //eprintln!("unknown table '{}'", table);
-    //}
-    //}
-    //}
-    //}
+    let path = args.out_path();
+    let raw_font = compiled.build_raw(&names).expect("ttf compile failed");
+    std::fs::write(path, raw_font).map_err(Into::into)
 }
 
+fn parse_glyph_order(path: &Path) -> Result<GlyphMap, Error> {
+    let contents = std::fs::read_to_string(path)?;
+    let map: GlyphMap = contents
+        .lines()
+        .filter(|l| !l.is_empty() && !l.starts_with('#'))
+        .map(|line| {
+            if line.bytes().any(|b| b.is_ascii_whitespace()) {
+                Err(Error::InvalidGlyphMap(format!(
+                    "name {line:?} contains whitespace"
+                )))
+            } else {
+                Ok(GlyphName::new(line))
+            }
+        })
+        .collect::<Result<_, _>>()?;
+    if map.get(".notdef") != Some(GlyphId::NOTDEF) {
+        Err(Error::InvalidGlyphMap("first glyph must be .notdef".into()))
+    } else {
+        Ok(map)
+    }
+}
+
+//FIXME: if we support having a font as input, again?
```

Comment:
Optional, delete since it's trivial to recover from history and once you get a bunch of this sort of thing it's really ugly